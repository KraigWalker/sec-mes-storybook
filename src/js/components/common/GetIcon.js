import React from "react";

const CommonGetIcon = (props) => {
  return (
    <svg
      role="img"
      fill={props.fill}
      className={props.className}
      viewBox={props.viewbox ? props.viewbox : "0 0 32 32"}
      width={props.width ? props.width : "32px"}
      height={props.height ? props.height : "32px"}
    >
      <use xlinkHref={`${window.baseURl}/images/icons.svg#${props.id}`} />
    </svg>
  );
};

export default CommonGetIcon;
