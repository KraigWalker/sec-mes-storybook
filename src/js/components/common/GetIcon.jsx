

import React from 'react';

class SvgIcon extends React.Component {
    constructor(props) {
		super(props);
		this.getContent = this.getContent.bind(this);
	}

    getContent(){
        let svgContent ;
       
        svgContent = <svg role="img" aria-hidden="true" className={this.props.className?this.props.className:""} viewBox="0 0 32 32" width={this.props.width ? this.props.width : "32px"} height={this.props.height ? this.props.height : "32px"}>
                <use xlinkHref={'../images/bow/bow-sprite.svg#' + this.props.id} />
            </svg>;
        
        return svgContent;
    }
    render() {
        return (
            this.getContent()
        );
    }
};
module.exports = SvgIcon;
