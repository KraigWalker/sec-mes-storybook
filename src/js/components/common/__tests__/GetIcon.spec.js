import { shallow } from 'enzyme';
import GetIcon from '../GetIcon';

describe('GetIcon Component Check', () => {
  const componentWrap = shallow(<GetIcon className="sampletext" width="24px" height="24px" id="someid" />);

  it('GetIcon component should mount', () => {
    expect(typeof componentWrap).toBe('object');
  });
  it('Props Check', () => {
    expect(componentWrap.props().className).toEqual('sampletext');
    expect(componentWrap.props().width).toEqual('24px');
    expect(componentWrap.props().height).toEqual('24px');
  });
  it('SVG URL verification', () => {
    expect(componentWrap.find("[xlinkHref='../../images/icons.svg#someid']")).toHaveLength(0);
  });
  it('should render correctly', () => {
    const tree = shallow(<GetIcon />);
    expect(tree).toMatchSnapshot();
  });
});
