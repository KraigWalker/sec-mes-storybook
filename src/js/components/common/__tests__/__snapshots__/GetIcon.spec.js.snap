// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GetIcon Component Check should render correctly 1`] = `ShallowWrapper {}`;
